advertisement/views.py

```diff
@@ -1,3 +1,152 @@
+from django.core.exceptions import ValidationError
+
+from user.utils  import advertiser_login_required, user_login_required, advertiser_login_mark, user_login_mark
+from user.models import User, Advertiser, InterestsType, State, Gender, UsersInterests
+from .models     import (AdvertisementCategory,Tag,  AdvertisementTag, Advertisement,
+                         AdvertisementLike, QuizHistory, AdvertisementsInterests 
+                        )
+
+class AdvertisementCreateListGetView(View):
+    @advertiser_login_required 
+    def post(self, request):
+        data = json.loads(request.body)
+   
+        try: 
+            new_advertisement = ( 
+                Advertisement.objects.create( 
+                    advertiser_id  = request.user.id,
+                    title          = data["title"],
+                    description    = data["description"],
+                    ad_category_id = data["ad_category_id"],
+                    video_link     = data["video_link"],
+                    thumbnail      = data["thumbnail"],
+                    budget         = data["budget"],
+                    price_per_view = data["price_per_view"],
+                )
+            )
+            
+            interests_bulk = [
+                    AdvertisementsInterests(
+                        advertisement_id  = new_advertisement.id,
+                        interests_type_id = interests
+                    ) for interests in data["interests_type_id"] 
+            ]     
+            AdvertisementsInterests.objects.bulk_create(interests_bulk) 
+
+            tag_list    = data["tag"]
+            tag_id_list = []
+            tags        = Tag.objects
+
+            for tag in tag_list: 
+                if tags.filter(name=tag).exists():
+                    tag_id_list.append(tags.filter(name=tag)[0].id)
+                else:
+                    new_tag = tags.create(name=tag)
+                    tag_id_list.append(new_tag.id) 
+            
+            tag_bulk = [
+                AdvertisementTag(
+                    advertisement_id = new_advertisement.id,
+                    tag_id           = tag_id
+                ) for tag_id in tag_id_list
+            ]
+            AdvertisementTag.objects.bulk_create(tag_bulk)
+            
+            return JsonResponse({"advertisement_id":new_advertisement.id}, status=200)
+    
+        except KeyError:
+            return JsonResponse({"ERROR":"KEY_MISSING"}, status=400)
+        except ValidationError:
+            return JsonResponse({"ERROR":"STRING_IN_NUMBER"}, status=400)
+
+    @user_login_mark
+    @advertiser_login_mark
+    def get(self, request):        
+        advertisements = Advertisement.objects.filter(deleted=False) 
+    
+        try:
+            offset         = int(request.GET.get('offset', '0'))
+            limit          = int(request.GET.get('limit', '3'))
+            category_id    = int(request.GET.get('category_id','0'))
```

Comment:
`category_id = int(request.GET['category_id']) if 'category_id' in request.GET else None`